### DISCUSSION TITLE
Filament Resource

### DISCUSSION THREAD

**Discussion by minvalencia** [filamentphp/filament#9829] [Help]
How to add form widget in resource?

**Comment by leandrocfe** [comment]
You could create a custom livewire component and [add a form](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component)

In the widget view file, you can render the livewire component 
```html
<div>
     @livewire('custom-form')
</div>
```

**Reply by minvalencia** [reply]
How can I added it in the resource, because I want to add it above the table?

**Reply by leandrocfe** [reply]
https://filamentphp.com/docs/3.x/panels/resources/widgets/#creating-a-resource-widget

https://filamentphp.com/docs/3.x/panels/resources/widgets/#displaying-a-widget-on-a-resource-page

You can use getHeaderWidgets()

**Reply by minvalencia** [reply]
What method I will use to call the form inside the custom widgets?

**Reply by leandrocfe** [reply]
Did you create a Livewire component?

https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component/#adding-the-form

**Reply by minvalencia** [reply]
I just create a simple resource.
![image](https://github.com/filamentphp/filament/assets/15870174/474a0c01-988a-489d-a4e3-2596c48341b6)

above the widget I want to create a custom form. It is possible to add in the resource page?

**Reply by leandrocfe** [reply]
1 - [create a livewire component and add a form](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component#setting-up-the-livewire-component)
2 - [create a widget to the resource](https://filamentphp.com/docs/3.x/panels/resources/widgets#creating-a-resource-widget)
3 - render the livewire component in the widget view `@livewire('my-custom-form')`

**Reply by minvalencia** [reply]
how to render the livewire in widget view in resource?

**Reply by leandrocfe** [reply]
```html
<div>
@livewire('my-custom-form')
</div>
```
?

**Reply by minvalencia** [reply]
This is my main resource
![image](https://github.com/filamentphp/filament/assets/15870174/485326d6-369b-4eb1-ac64-2f50e39963b6)
This is my widget
![image](https://github.com/filamentphp/filament/assets/15870174/b8865326-4127-46d2-b2ee-940728d07c30)
how can i include this
<div>
@livewire('my-custom-form')
</div>

**Reply by leandrocfe** [reply]
You should have a custom widget actually
https://filamentphp.com/docs/3.x/panels/resources/widgets#creating-a-resource-widget

**Reply by minvalencia** [reply]
Thanks for your help. I used renderhooks
->renderHook(
                'panels::page.header-widgets.before',
                fn (): string => Blade::render('@livewire(\'createtransaction\')'),
                scopes: \App\Filament\Resources\FightDetailResource\Pages\ManageFightDetails::class,
            )

**Reply by leandrocfe** [reply]
Should work as well
